packages/flutter_test/test/widget_tester_test.dart

```diff
@@ -821,6 +822,81 @@ void main() {
       binding.postTest();
     });
   });
+
+  group('Accessibility announcements testing API', () {
+    testWidgets('Returns last announcement', (WidgetTester tester) async {
+      await SemanticsService.announce('announcement 1', TextDirection.ltr);
+      await SemanticsService.announce('announcement 2', TextDirection.rtl,
+          assertiveness: Assertiveness.assertive);
+
+      final CapturedAccessibilityAnnouncement last = tester.getLastAnnouncement()!;
+      expect('announcement 2', equals(last.message));
+      expect(TextDirection.rtl, equals(last.textDirection));
+      expect(Assertiveness.assertive, equals(last.assertiveness));
+    });
+
+    testWidgets('Returns the list of announcements', (WidgetTester tester) async {
+      await SemanticsService.announce('announcement 1', TextDirection.ltr);
+      await SemanticsService.announce('announcement 2', TextDirection.rtl,
+          assertiveness: Assertiveness.assertive);
+      await SemanticsService.announce('announcement 3', TextDirection.rtl);
+
+      final List<CapturedAccessibilityAnnouncement> list = tester.takeAnnouncements()!;
+      expect(3, equals(list.length));
+      final CapturedAccessibilityAnnouncement first = list[0];
+      expect('announcement 1', equals(first.message));
+      expect(TextDirection.ltr, equals(first.textDirection));
+
+      final CapturedAccessibilityAnnouncement second = list[1];
+      expect('announcement 2', equals(second.message));
+      expect(TextDirection.rtl, equals(second.textDirection));
+      expect(Assertiveness.assertive, equals(second.assertiveness));
+
+      final CapturedAccessibilityAnnouncement third = list[2];
+      expect('announcement 3', equals(third.message));
+      expect(TextDirection.rtl, equals(third.textDirection));
+      expect(Assertiveness.polite, equals(third.assertiveness));
+
+      final List<CapturedAccessibilityAnnouncement>? emptyList = tester.takeAnnouncements();
+      expect(emptyList, isNull);
+    });
+
+    test('New test API is not breaking existing tests', () async {
```

Comment:
We do plan to break existing tests in the future by always specifying `assertiveness` in the protocol, right? So I'm not sure if this test provides real protection :thinking:

Reply:
My initial thought was to make sure that the current way of testing still works in general. Meaning that if someone wants to setup their own message handler, they're still able to do so. In another word, the new API doesn't overwrite what the user has specified as the message handler.
You had a good point about the future change though. So I've removed the announcements that didn't have `assertiveness` specified, since having only one would still accomplish what I had in mind. 
Let me know if you still think we don't need this test.